ddtrace/internal/telemetry/writer.py

```diff
@@ -237,7 +238,10 @@ def __init__(self, is_periodic=True, agentless=None):
         self._debug = asbool(os.environ.get("DD_TELEMETRY_DEBUG", "false"))
 
         self._enabled = config._telemetry_enabled
-        agentless = config._ci_visibility_agentless_enabled if agentless is None else agentless
+
+        if agentless is None:
+            agentless = config._ci_visibility_agentless_enabled or config._dd_api_key is not None
```

Comment:
Do we want to also check for `config._llmobs_agentless_enabled` here?

Reply:
oh snap, good catch - i must've forgotten to carry over the `_llmobs.py` file when I was splitting the PR 😬  ! check again - i've made another commit and just disabled the existing logic in `LLMObs.Enable`. I think it's a bit cleaner this way. lmk!